src/engine/objects/robot/SimRobotWheel.ts

```diff
@@ -136,22 +136,33 @@ export class SimRobotWheel extends SimObject {
     // Lateral
     // TODO the max lateral impulse should be proportional to mass
     const maxLateralImpulse = 2.0;
-    let impulse = this.getLateralVelocity().mul(
+    let impulseLateral = this.getLateralVelocity().mul(
       -this._body.getMass() * this._sideSlipFactor
     );
 
-    if (impulse.length() > maxLateralImpulse) {
-      impulse = impulse.mul(maxLateralImpulse / impulse.length());
+    if (impulseLateral.length() > maxLateralImpulse) {
+      impulseLateral = impulseLateral.mul(
+        maxLateralImpulse / impulseLateral.length()
+      );
     }
-    this._body.applyLinearImpulse(impulse, this._body.getWorldCenter());
+    this._body.applyLinearImpulse(impulseLateral, this._body.getWorldCenter());
 
     // Angular
     this._body.applyAngularImpulse(
       0.1 * this._body.getInertia() * -this._body.getAngularVelocity()
     );
 
     // Forward linear velocity
-    // TODO Implement Brake/Coast logic here
+    if (this._forceMagnitude < 0.000005) {
+      const impulseForward = this.getForwardVelocity()
```

Comment:
Nice change!

I'd probably rename `impulseForward` to something like `breakImpulse`?  Also maybe update the comment on line 155 that this code implements some sort of 'brake' logic?